### ISSUE TITLE
[Bug]: Hogwarp

### ISSUE THREAD

**Issue by trianl2** [parkervcp/eggs#2567]
### Panel Version

1.11.8

### Wings Version

1.11.8

### Service

Hogwarp

### Modified

No, I did not modify the egg

### Expected Behavior

Install the server but drop Nethost error

### Actual Behavior

Nethost error

### Steps To Reproduce

Step 1 Download the egg
Step 2 Make a server an install
Step 3 Drop Nethost error on startup

### Install logs

https://pteropaste.com/vzdj9ee9yb9z

**Comment by parkervcp** [comment]
Can you post the startup logs please

**Comment by QuintenQVD0** [comment]
![afbeelding](https://github.com/parkervcp/eggs/assets/67589015/cc3b6f78-3354-4fee-9f6b-10a10b4018f0)

is what it did for me and someone I know was able to join

**Comment by gOOvER** [comment]
the funny thing is; some games running with staging and have an output :/

**Comment by trianl2** [comment]
> ![afbeelding](https://user-images.githubusercontent.com/67589015/284020448-cc3b6f78-3354-4fee-9f6b-10a10b4018f0.png)
> 
> is what it did for me and someone I know was able to join

So that error doesnt matter? Thanks guys for answering so fast!

**Comment by gOOvER** [comment]
> > ![afbeelding](https://user-images.githubusercontent.com/67589015/284020448-cc3b6f78-3354-4fee-9f6b-10a10b4018f0.png)
> > is what it did for me and someone I know was able to join
> 
> So that error doesnt matter? Thanks guys for answering so fast!

can you connect? If no, then the error matter

And why the egg mark the server as started after the error?

**Comment by trianl2** [comment]
it needs https://dotnet.microsoft.com/en-us/download/dotnet/thank-you/runtime-desktop-7.0.5-windows-x64-installer to be installed, can someone modify the egg for that?

**Comment by gOOvER** [comment]
so winetrick_run must contain `dotnet7`

**Comment by trianl2** [comment]
![image](https://github.com/parkervcp/eggs/assets/48733937/b9fd86e2-710e-4585-bba6-282aff379260)
like that?

**Comment by trianl2** [comment]
![image](https://github.com/parkervcp/eggs/assets/48733937/7474c685-1451-4dc7-9973-53ebbdc78ee0)

**Comment by gOOvER** [comment]
looks like, the egg have an error. I dont own the game, so i cant test it

**Comment by QuintenQVD0** [comment]
> looks like, the egg have an error. I dont own the game, so i cant test it

I know the error was there. Winetricks does not yet support dotnet 7. But the huy i was making the egg with was able to join even with the dotnet error.

**Comment by gOOvER** [comment]
> > looks like, the egg have an error. I dont own the game, so i cant test it
> 
> I know the error was there. Winetricks does not yet support dotnet 7. But the huy i was making the egg with was able to join even with the dotnet error. 

hu? See His Screen. dotnet7 is supported

**Comment by QuintenQVD0** [comment]
> > > looks like, the egg have an error. I dont own the game, so i cant test it
> > 
> > 
> > I know the error was there. Winetricks does not yet support dotnet 7. But the huy i was making the egg with was able to join even with the dotnet error.
> 
> hu? See His Screen. dotnet7 is supported

Then they added it. But in testing then users were unable to join.

**Comment by gOOvER** [comment]
> > > > looks like, the egg have an error. I dont own the game, so i cant test it
> > > 
> > > 
> > > I know the error was there. Winetricks does not yet support dotnet 7. But the huy i was making the egg with was able to join even with the dotnet error.
> > 
> > hu? See His Screen. dotnet7 is supported
> 
> Then they added it. But in testing then users were unable to join.

7 months ago :)

**Comment by ImKringle** [comment]
Intended - this is the issue with plugins I mentioned *in the README*. Until Dotnet gets updated on Winetricks it gives a Nethost error. This doesnt do anything that affects the game, so it wasn't held off from use

Dotnet7 will break scripting and cause the loader to crash; hence why it's not there by default

@gOOvER Dotnet7 is in Winetricks on version 7.0.5 - these dlls specifically need something 7.0.11 or newer.

**Comment by ImKringle** [comment]
Until the winetricks gets a push or I decide to build a yolk with the right version being used; this is just a waiting game.

Doesnt break connection, and error is 100% ignorable. I've tested the egg and noted any problems it has lol

**Comment by trianl2** [comment]
> Until the winetricks gets a push or I decide to build a yolk with the right version being used; this is just a waiting game.
> 
> Doesnt break connection, and error is 100% ignorable. I've tested the egg and noted any problems it has lol

Yes, the only thing that is broken are the plugins

**Comment by ImKringle** [comment]
> > Until the winetricks gets a push or I decide to build a yolk with the right version being used; this is just a waiting game.
> > Doesnt break connection, and error is 100% ignorable. I've tested the egg and noted any problems it has lol
> 
> Yes, the only thing that is broken are the plugins

Yup. Read the README lol

**Comment by trianl2** [comment]
> > > Until the winetricks gets a push or I decide to build a yolk with the right version being used; this is just a waiting game.
> > > Doesnt break connection, and error is 100% ignorable. I've tested the egg and noted any problems it has lol
> > 
> > 
> > Yes, the only thing that is broken are the plugins
> 
> Yup. Read the README lol

I already know, but maybe this help the dev to solve it ^^ thanks to all

**Comment by ImKringle** [comment]
> > > ![afbeelding](https://user-images.githubusercontent.com/67589015/284020448-cc3b6f78-3354-4fee-9f6b-10a10b4018f0.png)
> > > is what it did for me and someone I know was able to join
> > 
> > 
> > So that error doesnt matter? Thanks guys for answering so fast!
> 
> can you connect? If no, then the error matter
> 
> And why the egg mark the server as started after the error?

This is just what I found was a constant and easily seen startup - Hogwarp only will show in the console if its showing something from it - and will crash pretty fast otherwise so it's the best bet for a startup. Could have had it watch for the port, but it does literally the same thing 🤷🏼

**Comment by trianl2** [comment]
> > > > ![afbeelding](https://user-images.githubusercontent.com/67589015/284020448-cc3b6f78-3354-4fee-9f6b-10a10b4018f0.png)
> > > > is what it did for me and someone I know was able to join
> > > 
> > > 
> > > So that error doesnt matter? Thanks guys for answering so fast!
> > 
> > 
> > can you connect? If no, then the error matter
> > And why the egg mark the server as started after the error?
> 
> This is just what I found was a constant and easily seen startup - Hogwarp only will show in the console if its showing something from it - and will crash pretty fast otherwise so it's the best bet for a startup. Could have had it watch for the port, but it does literally the same thing 🤷🏼
> 
> For a fix - Users would need a custom yolk that has the updated dotnet7 stuff in it at a minimum. Any winetrick / dll override _as is_ does nothing but break the egg more, hence why its blank by default

Can you share a custom yolk with dotnet7 stuff updated please Sir? hehe

**Comment by ImKringle** [comment]
> > > > > ![afbeelding](https://user-images.githubusercontent.com/67589015/284020448-cc3b6f78-3354-4fee-9f6b-10a10b4018f0.png)
> > > > > is what it did for me and someone I know was able to join
> > > > 
> > > > 
> > > > So that error doesnt matter? Thanks guys for answering so fast!
> > > 
> > > 
> > > can you connect? If no, then the error matter
> > > And why the egg mark the server as started after the error?
> > 
> > 
> > This is just what I found was a constant and easily seen startup - Hogwarp only will show in the console if its showing something from it - and will crash pretty fast otherwise so it's the best bet for a startup. Could have had it watch for the port, but it does literally the same thing 🤷🏼
> > For a fix - Users would need a custom yolk that has the updated dotnet7 stuff in it at a minimum. Any winetrick / dll override _as is_ does nothing but break the egg more, hence why its blank by default
> 
> Can you share a custom yolk with dotnet7 stuff updated please Sir? hehe

I haven't made one yet, as yolks/dockers are something I'm still figuring out - Once I get one working this obviously will get an update, but thats really a TBD since its not a priority currently

Anyone is free to make one and update this or provide one, just not something im in much of a hurry to get working lol The egg is more of a side project when I have time

**Comment by trianl2** [comment]
> > > > > > ![afbeelding](https://user-images.githubusercontent.com/67589015/284020448-cc3b6f78-3354-4fee-9f6b-10a10b4018f0.png)
> > > > > > is what it did for me and someone I know was able to join
> > > > > 
> > > > > 
> > > > > So that error doesnt matter? Thanks guys for answering so fast!
> > > > 
> > > > 
> > > > can you connect? If no, then the error matter
> > > > And why the egg mark the server as started after the error?
> > > 
> > > 
> > > This is just what I found was a constant and easily seen startup - Hogwarp only will show in the console if its showing something from it - and will crash pretty fast otherwise so it's the best bet for a startup. Could have had it watch for the port, but it does literally the same thing 🤷🏼
> > > For a fix - Users would need a custom yolk that has the updated dotnet7 stuff in it at a minimum. Any winetrick / dll override _as is_ does nothing but break the egg more, hence why its blank by default
> > 
> > 
> > Can you share a custom yolk with dotnet7 stuff updated please Sir? hehe
> 
> I haven't made one yet, as yolks/dockers are something I'm still figuring out - Once I get one working this obviously will get an update, but thats really a TBD since its not a priority currently
> 
> Anyone is free to make one and update this or provide one, just not something im in much of a hurry to get working lol The egg is more of a side project when I have time

Yes we totally understand, and we all appreciate all the work you have done guys, you are awesome and I am in love with this community.

**Comment by QuintenQVD0** [comment]
> > > > > > ![afbeelding](https://user-images.githubusercontent.com/67589015/284020448-cc3b6f78-3354-4fee-9f6b-10a10b4018f0.png)
> > > > > > is what it did for me and someone I know was able to join
> > > > > 
> > > > > 
> > > > > So that error doesnt matter? Thanks guys for answering so fast!
> > > > 
> > > > 
> > > > can you connect? If no, then the error matter
> > > > And why the egg mark the server as started after the error?
> > > 
> > > 
> > > This is just what I found was a constant and easily seen startup - Hogwarp only will show in the console if its showing something from it - and will crash pretty fast otherwise so it's the best bet for a startup. Could have had it watch for the port, but it does literally the same thing 🤷🏼
> > > For a fix - Users would need a custom yolk that has the updated dotnet7 stuff in it at a minimum. Any winetrick / dll override _as is_ does nothing but break the egg more, hence why its blank by default
> > 
> > 
> > Can you share a custom yolk with dotnet7 stuff updated please Sir? hehe
> 
> I haven't made one yet, as yolks/dockers are something I'm still figuring out - Once I get one working this obviously will get an update, but thats really a TBD since its not a priority currently
> 
> Anyone is free to make one and update this or provide one, just not something im in much of a hurry to get working lol The egg is more of a side project when I have time

Will in no way work. You can not just install linux dotnet and expect that it works.  Or windows dotnet. This must be fixt in winetricks and can not be fixt on out side in a docker image.

**Comment by ImKringle** [comment]
Quinten tried a few tricks to get this running (even manually updating winetricks) and no dice, this seems like it may just be a wait on a linux release when the mod is *eventually* finished.

**Comment by gOOvER** [comment]
> Quinten tried a few tricks to get this running (even manually updating winetricks) and no dice, this seems like it may just be a wait on a linux release when the mod is _eventually_ finished.

why updating winetricks manually? It will be updated every week

**Comment by QuintenQVD0** [comment]
> > Quinten tried a few tricks to get this running (even manually updating winetricks) and no dice, this seems like it may just be a wait on a linux release when the mod is _eventually_ finished.
> 
> why updating winetricks manually? It will be updated every week

I update in winetricks the dotnet7 version from 7.0.4 to 7.0.14 as they ention they need 7.0.13 to run it but it still does not work.

**Comment by gOOvER** [comment]
> > > Quinten tried a few tricks to get this running (even manually updating winetricks) and no dice, this seems like it may just be a wait on a linux release when the mod is _eventually_ finished.
> > 
> > 
> > why updating winetricks manually? It will be updated every week
> 
> I update in winetricks the dotnet7 version from 7.0.4 to 7.0.14 as they ention they need 7.0.13 to run it but it still does not work.

https://github.com/Winetricks/winetricks/pull/2149/files

**Comment by QuintenQVD0** [comment]
Closed as of #2576 , the egg works just can not load plugin because there dotnet dll do not work with wine
